components/weather_data/WeatherStationMap.tsx

```diff
@@ -107,17 +107,13 @@ export const WeatherStationMap: React.FunctionComponent<{
   }, [avalancheCenterMapRegion, controller]);
 
   // we want light grey zones in the background here
-  const zones: MapViewZone[] = mapLayer.features.map((feature: MapLayerFeature) => ({
-    zone_id: feature.id,
-    geometry: feature.geometry,
-    hasWarning: false,
-    center_id: center_id,
-    name: feature.properties.name,
-    danger_level: DangerLevel.None,
-    start_date: feature.properties.start_date,
-    end_date: feature.properties.end_date,
-    fillOpacity: 0.1,
-  }));
+  const zones: MapViewZone[] = mapLayer.features.map((feature: MapLayerFeature) => {
+    const zone = MapViewZoneFor(center_id, feature);
+    zone.hasWarning = false;
+    zone.danger_level = DangerLevel.None;
+    zone.fillOpacity = 0.1;
+    return zone;
+  });
```

Comment:
you could get more terse here and eliminate the explicit return 😁. I think reasonable people could disagree on that one though - totally up to you.

```suggestion
  const zones: MapViewZone[] = mapLayer.features.map((feature: MapLayerFeature) => ({
    ...mapViewZoneFor(center_id, feature),
    hasWarning: false,
    danger_level: DangerLevel.None,
    fillOpacity: 0.1,
  }));
```

Reply:
It looks like it would on the surface, but I didn't know if that was the sort of thing that would get JIT-ed out or not! Also was keen to hear if you thought it was more of an academic concern or would have any real impact on the app, thanks for sharing your thoughts!